fluss-common/src/main/java/org/apache/fluss/lake/source/LakeSource.java

```diff
@@ -69,10 +69,12 @@ public interface LakeSource<Split extends LakeSplit> extends Serializable {
      * Creates a record reader for reading data from the lake source for the specified split.
      *
      * @param context The reader context containing the split to be read
+     * @param readInOrder Whether the reader should read lake data in order
      * @return A record reader instance for the given split
      * @throws IOException if an error occurs during reader creation
      */
-    RecordReader createRecordReader(ReaderContext<Split> context) throws IOException;
+    RecordReader createRecordReader(ReaderContext<Split> context, boolean readInOrder)
+            throws IOException;
```

Comment:
We don't always need to create a SortedReader. For primary key tables in streaming mode, a regular RecordReader is sufficient.